operators/derive/to_es.py

```diff
@@ -1,38 +1,67 @@
 from srm_tools.logger import logger
 
 
-def set_es_item_types(resources=None):
-    def func(package):
-        matcher = ResourceMatcher(resources, package.pkg)
-        for resource in package.pkg.descriptor['resources']:
-            if matcher.match(resource['name']):
-                for field in resource["schema"]["fields"]:
-                    if field["type"] == "any":
-                        field["type"] = "string"
-                    elif field["type"] == "array":
-                        field["es:itemType"] = "string"
-        yield package.pkg
-
-        res_iter = iter(package)
-        for r in res_iter:
-            if matcher.match(r.res.name):
-                yield r.it
-            else:
-                yield r
-
-    return func
+class SRMMappingGenerator(MappingGenerator):
+    @classmethod
+    def _convert_type(cls, schema_type, field, prefix):
+        # TODO: should be in base class
+        if field['type'] == 'any':
+            field['es:itemType'] = 'string'
+        prop = super()._convert_type(schema_type, field, prefix)
+        boost, keyword = field.get('es:boost'), field.get('es:keyword')
+        if keyword:
+            prop['type'] = 'keyword'
+        if boost:
+            prop['boost'] = boost
+        if schema_type in ('number', 'integer', 'geopoint'):
+            prop['index'] = True
+        return prop
 
 
 def data_api_es_flow():
     return DF.Flow(
-        DF.load(f'{settings.DATA_DUMP_DIR}/table_data/datapackage.json'),
+        DF.load(f'{settings.DATA_DUMP_DIR}/card_data/datapackage.json'),
+        DF.set_type('card_id', type='string', **{'es:keyword': True}),
+        DF.set_type('branch_id', type='string', **{'es:keyword': True}),
+        DF.set_type('service_id', type='string', **{'es:keyword': True}),
+        DF.set_type('organization_id', type='string', **{'es:keyword': True}),
+        DF.set_type(
+            'response_categories', type='array', **{'es:itemType': 'string', 'es:keyword': True}
+        ),
+        DF.set_type(
+            'situations',
+            type='array',
+            **{
+                'es:itemType': 'object',
+                'es:schema': {
+                    'fields': [
+                        {'type': 'string', 'name': 'id', 'es:keyword': True},
+                        {'type': 'string', 'name': 'name'},
+                    ]
+                },
+            },
+        ),
+        DF.set_type(
+            'responses',
+            type='array',
+            **{
+                'es:itemType': 'object',
+                'es:schema': {
+                    'fields': [
+                        {'type': 'string', 'name': 'id', 'es:keyword': True},
+                        {'type': 'string', 'name': 'name'},
+                    ]
+                },
+            },
+        ),
         dump_to_es(
-            indexes=dict(srm_api=[dict(resource_name='table_data')]),
+            indexes=dict(srm_api=[dict(resource_name='card_data')]),
```

Comment:
let's call this index 'cards' as we have one index per resource

Reply:
ok.